### PR TITLE
Add ability of customizing file name

### PR DESCRIPTION
**Pull request by sleagon**
Just as mentioned in issue #72 which was opened more than one month ago, it is useful to customlize the name of target file.

Usage:

```bash
mkcert --key-file my-cert.pem  --cert-file my-key.pem example.com '*.example.org' myapp.dev localhost 127.0.0.1 ::1
```

or for p12 file

```bash
mkcert --p12-file my.p12 example.com '*.example.org' myapp.dev localhost 127.0.0.1 ::1
```

### CODE REVIEW

File: cert.go [hunk 1]
```diff
@@ -39,6 +40,34 @@ func init() {
 	userAndHostname += strings.TrimSpace(string(out))
 }
 
+// getFileName generate file name according to flags
+func (m *mkcert) getFileName(w string, args []string) (name string, err error) {
+	filename := strings.Replace(args[0], ":", "_", -1)
+	filename = strings.Replace(filename, "*", "_wildcard", -1)
```
Comment (adamdecaf) [inline cert.go]: Are these replacements needed because of how we generate names?

Comment (sleagon) [reply cert.go]: Yes, I do not want to modify the origin principle, if *FileFlag is not supplied, default filename will be generated, which means this PR has no effects on the users who do not care about the target file name.

Comment (adamdecaf) [reply cert.go]: Ok, cool. Can we write a test around `getFileName`?

Comment (sleagon) [reply cert.go]: > Ok, cool. Can we write a test around `getFileName`?

Test cases added. @adamdecaf

**Comment by sleagon** [comment]
> mkcert --key-file ./foo/key.pem --cert-file ./foo/cert.pem localhost

Actually, providing dir will not break anything, your script failed because directory foo does not exist. I do not think we should  do anything like ```mkdir -p``` to fix this. Just try to mkdir foo and rerun this script.

**Comment by adamdecaf** [comment]
> Just try to mkdir foo and rerun this script.

Whoops, forgot to mention that!

**Comment by AlekSi** [comment]
@FiloSottile PTAL

**Comment by annaswims** [comment]
This is great!  Puma doesn't like a "+" in the file name and this would take care of that issue more gracefully.

File: cert.go [hunk 2]
```diff
@@ -111,9 +140,9 @@ func (m *mkcert) makeCert(hosts []string) {
 	}
 
 	if !m.pkcs12 {
-		log.Printf("\nThe certificate is at \"./%s.pem\" and the key at \"./%s-key.pem\" ✅\n\n", filename, filename)
+		log.Printf("\nThe certificate is at \"./%s\" and the key at \"./%s\" ✅\n\n", keyname, certname)
```
Comment (Quintinity) [inline cert.go]: Shouldn't the parameters be the other way around?

Comment (sleagon) [reply cert.go]: Yes, thank you very much for this, i'll fix it right now.

Comment (Quintinity) [reply cert.go]: Great! Thanks for this PR - I hope it gets merged soon.

**Comment by FiloSottile** [comment]
Thank you for the PR, I merged it with a few minor changes.

**Pull request merged**